### PR TITLE
test: only submit extensions to proposer

### PR DESCRIPTION
**Pull request by aljo242**

### CODE REVIEW

File: consensus/reactor.go [hunk 1]
```diff
@@ -851,29 +871,29 @@ func (conR *Reactor) gossipVotesForHeight(
...
-		if ps.PickSendVote(rs.Votes.Prevotes(prs.Round)) {
+		if ps.PickSendVote(rs.Votes.Prevotes(prs.Round), true) {
 			logger.Debug("Picked rs.Prevotes(prs.Round) to send", "round", prs.Round)
 			return true
 		}
 	}
 	// If there are precommits to send...
 	if prs.Step <= cstypes.RoundStepPrecommitWait && prs.Round != -1 && prs.Round <= rs.Round {
-		if ps.PickSendVote(rs.Votes.Precommits(prs.Round)) {
+		if ps.PickSendVote(rs.Votes.Precommits(prs.Round), sendExtensions) {
```
Comment (aljo242) [inline consensus/reactor.go:889]: We only care about it in the pre-commit phase right?

File: consensus/reactor.go [hunk 2]
```diff
@@ -759,21 +768,31 @@ OUTER_LOOP:
 			sleeping = 0
 		}
 
+		// if the proposer is in our peerlist && proposer is the function peer, send the extensions
+		// if the proposer is in our peerlist && proposer is not function peer, do not send the extensions
+		// else if the proposer is not in our peerlist, send extensions to all
+		sendExtensions := true
+		proposerPeerID := p2p.PubKeyToID(rs.Validators.Proposer.PubKey)
+		if conR.Switch.IsPeerUnconditional(proposerPeerID) {
```
Comment (g3ndo9kar9) [inline consensus/reactor.go]: Why are we checking if the peer is unconditional?

File: consensus/reactor.go [hunk 3]
```diff
@@ -733,6 +736,12 @@ func (conR *Reactor) gossipVotesRoutine(peer p2p.Peer, ps *PeerState) {
 	// Simple hack to throttle logs upon sleep.
 	sleeping := 0
 
+	consAddrHexBz, err := hex.DecodeString(string(peer.ID()))
```
Comment (g3ndo9kar9) [inline consensus/reactor.go:743]: nit: would move logic to a helper

**Review by MaghnusM — APPROVED** [review]
Seems good tbh

**Review by g3ndo9kar9 — COMMENTED** [review]
question

File: consensus/reactor.go (hunk 2 above)
Comment (aljo242) [reply consensus/reactor.go]: https://github.com/skip-mev/cometbft/pull/1/commits/3acd630aa330c2514a11d40a67dfd788662ca31f

File: consensus/reactor.go (hunk 1 above)
Comment (g3ndo9kar9) [reply consensus/reactor.go:889]: Ya only pre-commits have a VE attached

File: consensus/state.go [hunk 4]
```diff
@@ -2175,7 +2175,12 @@ func (cs *State) addVote(vote *types.Vote, peerID p2p.ID) (added bool, err error
 		// Verify VoteExtension if precommit and not nil
 		// https://github.com/tendermint/tendermint/issues/8487
 		if vote.Type == cmtproto.PrecommitType && !vote.BlockID.IsZero() &&
-			!bytes.Equal(vote.ValidatorAddress, myAddr) { // Skip the VerifyVoteExtension call if the vote was issued by this validator.
+			!bytes.Equal(vote.ValidatorAddress, myAddr) &&
+			bytes.Equal(cs.Validators.Proposer.Address, myAddr) &&
```
Comment (aljo242) [inline consensus/state.go]: This should also be `Nextvalidators.Proposer` right?

File: consensus/state.go [hunk 5]
```diff
@@ -2175,7 +2175,12 @@ func (cs *State) addVote(vote *types.Vote, peerID p2p.ID) (added bool, err error
 		// Verify VoteExtension if precommit and not nil
 		// https://github.com/tendermint/tendermint/issues/8487
 		if vote.Type == cmtproto.PrecommitType && !vote.BlockID.IsZero() &&
-			!bytes.Equal(vote.ValidatorAddress, myAddr) { // Skip the VerifyVoteExtension call if the vote was issued by this validator.
+			!bytes.Equal(vote.ValidatorAddress, myAddr) &&
+			bytes.Equal(cs.state.NextValidators.Proposer.Address, myAddr) &&
```
Comment (aljo242) [inline consensus/state.go:2179]: this should be `Nextvalidators` right?

Comment (g3ndo9kar9) [reply consensus/state.go:2179]: yep

**Pull request closed without merging**